src/app/clusters/ota-requestor/OTARequestor.cpp

```diff
@@ -118,6 +118,9 @@ void OTARequestor::OnQueryImageResponse(void * context, const QueryImageResponse
 {
     LogQueryImageResponse(response);
 
+    uint32_t currentSoftwareVersion;
+    VerifyOrDie(Basic::Attributes::SoftwareVersion::Get(kRootEndpointId, &currentSoftwareVersion) == EMBER_ZCL_STATUS_SUCCESS);
```

Comment:
`VerifyOrDie` seems too drastic here. Do we really need to abort everything if we can't read the current version? I don't know if there's any recourse we can take here, but we can at least just not go ahead with the update. Recommend using `VerifyOrReturn` or something similar.

I see now that there are existing `VerifyOrDie` calls here. @carol-apple @selissia are these really necessary?

Reply:
I agree, VerifyOrDie() is too drastic

Reply:
@Damian-Nordic suggested to abort

Reply:
FYI, current software version is already cached at https://github.com/project-chip/connectedhomeip/blob/master/src/app/clusters/ota-requestor/OTARequestor.h#L90

Reply:
> are these really necessary?

Aren't we in some pretty weird state if we can't read Basic attributes? Is there a point to continue?